proto/viam/app/v1/app.proto

```diff
@@ -810,6 +810,8 @@ message Fragment {
   FragmentVisibility visibility = 12;
   // latest timestamp when fragment was updated
   google.protobuf.Timestamp last_updated = 13 [(tagger.v1.tags) = "bson:\"last_updated_at\""];
+  // number of machines in the requester org using the fragment
+  int32 machines_in_org_count = 14;
```

Comment:
Agreed with @mcous - your concern is valid, and hoisting usage to the `GetFragmentResponse` message level avoids coupling usage with the fragment itself.

I don't know what your anticipated usage is like, but I think there's an argument to be made that this should be a separate API to keep the usage completely decoupled from the fragment itself. Something like this:

```
message GetFragmentUsageRequest {
  string organization_id = 1;
}

message GetFragmentUsageResponse {
  FragmentUsage usage = 1;
}
```

Decoupling further would avoid any breaking changes to the existing API as well. You would need to think about whether or not you need usage in `ListFragments` as well.

Reply:
this looks good to me. should we use `repeated FragmentUsage` instead of a map here for additional flexibility? thinking about fragment versions, i think `repeated` might just be better supported in proto as well (preserved ordering)

```proto
message FragmentUsage {
  int32 fragment_id = 1;
  int32 machines_in_org_count = 2;
}

message ListFragmentsResponse {
  repeated Fragment fragments = 1;
  repeated FragmentUsage usage_by_fragment_id = 2;
}
```

Reply:
👍 `repeated` makes sense to me

Reply:
@ethanlookpotts fyi, we do need usage in `ListFragments` as well. would you still recommend creating a separate endpoint? we would have to duplicate fragment query-ing logic (visibility, by org, show public, etc) or wait for a list of fragment IDs from `ListFragments`

Reply:
thank you all for the opinions!

Reply:
In terms of `map` vs `repeated` - is ordering important? I would expect you are iterating over fragments, and would want O(1) lookup of the usage.

Reply:
> In terms of `map` vs `repeated` - is ordering important? I would expect you are iterating over `fragments`, and would want O(1) lookup of `usage_by_fragment_id`.

Reply:
i imagine we'd convert to a map on the frontend, the flexibility of indexing from `repeated` messages seems like a worthy tradeoff to me

Reply:
What do you mean by flexibility of indexing? It'd be nice to not need to convert to a map on the frontend.

(I don't feel strongly, I'm just curious 😄)

Reply:
with fragment versions, we will need two pieces of data to index fragment usage: a fragment ID and a fragment version. idk how we are planning to adapt everything to this yet, but i imagine it could be nice to avoid converting to some composite string in the protos and instead do so somewhere more adaptable (the frontend)

Reply:
Gotcha - I would personally keep the map and eventually make the fragment usage be this, but it's ultimately your call if you prefer the repeated field!

```
message FragmentUsage {
  map<string, int32> machines_count_by_version = 2;

  reserved 1;
  reserved 'machines_in_org_count';
}
```

Reply:
I crafting a proto suggestion that is similar what Ethan suggested. Having a separate API initially seems like overkill but it helps follow the request of a specific org_id and expecting the results to match what is requested. Less ambiguity with this approach, and can support 1+ :

```
message GetFragmentsUsageRequest {
  string organization_id = 1;
  repeated string fragment_ids = 1;
}

message GetFragmentUsageResponse {
  repeated FragmentUsage usage = 1;
}

message FragmentUsage {
  string fragment_id = 1;
  int64 machines_in_org_count = 2;
}
```

I know that probably adds some complexity on the frontend? Having to interleave results from two APIs into one table.

Reply:
i think the bigger problem with this is we will have to wait for the `ListFragments` request to return to supply the `fragment_ids` to this endpoint, so fragment usage stats would display delayed

Reply:
I agree with Emily, we need to avoid request waterfalls if we separate out the endpoints. I'm amendable to two requests if we can fire them off concurrently, but I still feel like one request is a better UX

Reply:
Ok yeah I figured this. Let's not do another API. Now I'm thinking similar to Michael's comment above which is to move and rename all usage stats from the Fragment message to the Usage message. Eventually deprecating the existing ones in the Fragment message:

```
message FragmentUsage {
  int64 total_organizations = 1;
  int64 total_machines = 2;
  int64 total_parts = 3;
  int64 current_organization_machines = 4;
...
}
```